v3/src/components/graph/utilities/graph-utils.ts

```diff
@@ -675,3 +679,49 @@ export const curveBasis = (points: Point[]) => {
   path += pathBasis(p2, p3, p3, p3)
   return path
 }
+
+const axisPlaceToOrientationMap: Record<GraphPlace, AxisOrientation> = {
+  "bottom": "horizontal",
+  "top": "horizontal",
+  "left": "vertical",
+  "rightCat": "vertical",
+  "rightNumeric": "vertical",
+  "yPlus": "vertical",
+  "plot": "horizontal",
+  "legend": "horizontal"
+}
+
+export interface IUpdateGraphValues {
+  attributeId: string | number
+  attributeName?: string
+  axisOrientation?: string
+  plotType?: string,
+  primaryRole?: string
+}
+
+export const attrChangeNotificationValues = (
+  place: GraphPlace, attrId: string, attrName?: string, attrIdToRemove = "",
+  tile?: ITileModel
+) => {
+
+  const axisOrientation = axisPlaceToOrientationMap[place] || "axis"
+
+  const attributeName = attrIdToRemove
+    ? `Remove ${ axisOrientation === "horizontal" ? "X" : "Y" }: ${attrName}`
+    : attrName
+
+  const graphContent = tile?.content as IGraphContentModel
+  const plotType = graphContent?.plotType
+  const primaryRole = graphContent?.dataConfiguration.primaryRole || "x"
```

Comment:
I'm especially unsure of including these values here. They're not included in V2 notifications about attribute changes. 

The information will be useful for the DAVAI plugin. I'm not sure how else to send notifications about changes to those things aside from including them here. I think the plugin could make a follow-up `get` request for that information after receiving an attribute change notification. From the plugin's perspective it'd be better to not have to do that. I'm not sure that's a good enough reason to add this information here, though.

Reply:
I don't have a problem with returning additional information if it's helpful. In #1910 the convention established is that the property returned is named `primaryAxis`, so we should maintain that convention here.